index.js

```diff
@@ -49,73 +132,99 @@ const shape = (columns, opt) => (data, i) => {
+	captionsPosition: 1.2, // radial position of captions
+	smoothing: noSmoothing, // default smoothing function
+	axisProps: () => ({className: 'axis'}), // default axis properties
+	scaleProps: () => ({className: 'scale', fill: 'none'}), // default scale properties
+	shapeProps: () => ({className: 'shape'}), // default shape properties
+	captionProps: () => ({ // default caption properties
 		className: 'caption',
 		textAnchor: 'middle', fontSize: 3,
 		fontFamily: 'sans-serif'
 	})
-}
-
-const renderRadarChart = (columns, data, opt = {}) => {
-	if ('object' !== typeof columns || Array.isArray(columns)) {
+})
+
+// Main function to render the radar chart with given columns and data
+/**
+ * @template {Exclude<string, 'class'>} T
+ * @param {Record<Exclude<T, 'class'>, string>} columnsData - The data for the columns.
+ * @param {Data<T>[]} data - The array of data to be plotted.
+ * @param {Partial<Options<T>>} opt - Additional options to override the defaults.
+ * @returns {string} The rendered radar chart as an SVG element.
+ */
+const renderRadarChart = (columnsData, data, opt = {}) => {
+	if (typeof columnsData !== 'object' || Array.isArray(columnsData)) {
 		throw new Error('columns must be an object')
 	}
 	if (!Array.isArray(data)) {
 		throw new Error('data must be an array')
 	}
-	opt = Object.assign({}, defaults, opt)
-	opt.chartSize = opt.size / opt.captionsPosition
+	if (data.some(data => Object.keys(data).some(key => key !== 'class' && typeof data[key] !== 'number'))) {
+		throw new Error('data must contain set of numbers')
+	}
+	const options = /** @type {ExtendedOptions<T>} */({...defaults, ...opt, chartSize: 0})
+	options.chartSize = options.size / options.captionsPosition
 
-	columns = Object.keys(columns).map((key, i, all) => ({
-		key, caption: columns[key],
+	const columns = /** @type {Column<T>[]} */ (Object.keys(columnsData).map((key, i, all) => ({
+		key, caption: columnsData[key],
 		angle: Math.PI * 2 * i / all.length
-	}))
+	})))
 
 	const groups = [
-		h('g', data.map(shape(columns, opt)))
+		h('g', data.map(shape(columns, options)))
 	]
-	if (opt.captions) groups.push(h('g', columns.map(caption(opt))))
-	if (opt.axes) groups.unshift(h('g', columns.map(axis(opt))))
-	if (opt.scales > 0) {
+	if (options.captions) groups.push(h('g', columns.map(caption(options))))
+	if (options.axes) groups.unshift(h('g', columns.map(axis(options))))
+	if (options.scales > 0) {
 		const scales = []
-		for (let i = opt.scales; i > 0; i--) {
-			scales.push(scale(opt, i / opt.scales))
+		for (let i = options.scales; i > 0; i--) {
+			scales.push(scale(options, i / options.scales))
 		}
 		groups.unshift(h('g', scales))
 	}
 
-	const delta = (opt.size / 2).toFixed(4)
+	const delta = (options.size / 2).toFixed(4)
 	return h('g', {
 		transform: `translate(${delta},${delta})`
 	}, groups)
 }
 
+// Export the radar chart rendering function
 export {
 	renderRadarChart as radar,
 }
+
+export * from './smoothing.js'
```

Comment:
Please revert this change. I want to keep the smoothing part completely separate to allow smaller bundle sizes (when no tree-shaking-like processing is in place).

Reply:
Done